lib/recipe.rb

```diff
@@ -0,0 +1,151 @@
+#  Copyright 2017 Chef Software, Inc.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+liveness_agent = <<'AUTOMATE_LIVENESS_AGENT'
+#LIVENESS_AGENT
+AUTOMATE_LIVENESS_AGENT
+
+liveness_agent.gsub!('#!/usr/bin/env ruby', "#!#{Gem.ruby}")
+
+agent_dir     = '/var/opt/chef/'
+agent_bin_dir = ::File.join(agent_dir, 'bin')
+agent_etc_dir = ::File.join(agent_dir, 'etc')
+agent_log_dir = '/var/log/chef'
+agent_bin     = ::File.join(agent_bin_dir, 'automate-liveness-agent')
+agent_conf    = ::File.join(agent_etc_dir, 'config.json')
+server_uri    = URI(Chef::Config[:chef_server_url])
+
+[agent_bin_dir, agent_etc_dir, agent_log_dir].each do |dir|
+  directory dir do
+    recursive true
+  end
+end
+
+file agent_bin do
+  mode 0755
+  owner 'root'
+  group 'root'
+  content liveness_agent
+end
+
+
+file agent_conf do
+  mode 0755
+  owner 'root'
+  group 'root'
+
+  content(
+    lazy do
+      config = {
+        'client_key_path' => Chef::Config[:client_key],
+        'client_name' => node.name,
+        'chef_server_fqdn' => server_uri.host,
+        'data_collector_url' => Chef::Config[:data_collector][:server_url],
+        'org_name' => Chef::Config[:data_collector][:organization] || server_uri.path.split('/').last,
+        'entity_uuid' => Chef::JSONCompat.parse(
+          Chef::FileCache.load('data_collector_metadata.json')
+        )['node_uuid'],
+        'unprivileged_uid' => 100,
+        'unprivileged_gid' => 200,
```

Comment:
Might want to set these to `nil` for now, since the privilege change stuff is merged and this will get acted on.